cloudant-http/src/main/java/com/cloudant/http/interceptors/Replay429Interceptor.java

```diff
@@ -0,0 +1,160 @@
+     *
+     * @param numberOfReplays  number of times to replay a request that received a 429
+     * @param initialBackoff   the initial delay before retrying
+     * @param preferRetryAfter whether the replay should honour the duration specified by a
+     *                         Retry-After header sent by the server in preference to the local
+     *                         doubling backoff.
+     */
+    public Replay429Interceptor(int numberOfReplays, long initialBackoff, boolean
+            preferRetryAfter) {
+        this.numberOfReplays = numberOfReplays;
+        this.initialSleep = initialBackoff;
+        this.preferRetryAfter = preferRetryAfter;
+    }
+
+    @Override
+    public HttpConnectionInterceptorContext interceptResponse(HttpConnectionInterceptorContext
+                                                                      context) {
+
+        // Get or init the stored context for this interceptor
+
+        try {
+            HttpURLConnection urlConnection = context.connection.getConnection();
+            int code = urlConnection.getResponseCode();
+
+            // We only want to take action on a 429 response
+            if (code != 429) {
+                return context;
+            }
+
+            // We received a 429
+
+            // Get the counter from the request context state
+            AtomicInteger attemptCounter = context.getState(this, "attempt", AtomicInteger.class);
+
+            // If there was no counter yet, then this is the first 429 received for this request
+            if (attemptCounter == null) {
+                context.setState(this, "attempt", (attemptCounter = new AtomicInteger()));
+            }
+
+            // Get the current value, and then increment for the next time round
+            int attempt = attemptCounter.getAndIncrement();
+
+            // Check if we have remaining replays
+            if (attempt < numberOfReplays && context.connection.getNumberOfRetriesRemaining() > 0) {
+
+                // If the response includes a Retry-After then that is when we will retry, otherwise
+                // we use a doubling sleep
+                long sleepTime = 0l;
+
+                String retryAfter = (preferRetryAfter) ? urlConnection.getHeaderField
+                        ("Retry-After") : null;
+                if (retryAfter != null) {
+                    // See https://tools.ietf.org/html/rfc6585#section-4
+                    // Whilst not specified for 429 for 3xx or 503 responses the Retry-After header
+                    // is expressed as an integer number of seconds or a date in one of the 3 HTTP
+                    // date formats https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.3
+                    // Cloudant servers should give us an integer number of seconds, so don't worry
+                    // about parsing dates for now.
+                    sleepTime = Long.parseLong(retryAfter) * 1000;
```

Comment:
Possibly need to sanity check `Retry-After` - what if it isn't a valid number or it's absurdly large? A naughty server could cause us to hang by making us sleep for a really long time.